static/app/views/issueDetails/groupDetails.tsx

```diff
@@ -269,14 +281,33 @@ function useFetchGroupDetails({
   const allGroups = useLegacyStore(GroupStore);
   const group = allGroups.find(({id}) => id === params.groupId) as Group;
 
-  const [loading, setLoading] = useState<boolean>(false);
   const [project, setProject] = useState<Project | null>(null);
-  const [event, setEvent] = useState<Event | null>(null);
   const [loadingGroup, setLoadingGroup] = useState<boolean>(false);
-  const [loadingEvent, setLoadingEvent] = useState<boolean>(false);
   const [error, setError] = useState<boolean>(false);
   const [errorType, setErrorType] = useState<Error | null>(null);
-  const [eventError, setEventError] = useState<boolean>(false);
+  const [event, setEvent] = useState<Event | null>(null);
+
+  const groupId = params.groupId;
+  const eventId = params.eventId ?? 'latest';
+
+  const eventUrl = `/issues/${groupId}/events/${eventId}/`;
+
+  const eventQuery: {environment?: string[]} = {};
+  if (environments.length !== 0) {
+    eventQuery.environment = environments;
+  }
+
+  const {
+    data: eventData,
```

Comment:
You will also need to use `refetch` which is returned from this and use it in the `refetchData` function. Otherwise the retry button here won't refetch the event:

![CleanShot 2023-05-09 at 13 38 34](https://github.com/getsentry/sentry/assets/10888943/d07b64cb-6b30-4867-9852-f62535354ecc)